pkg/rabbitmq/cluster.go

```diff
@@ -0,0 +1,284 @@
+			// for the random user in /etc/passwd has the correct homedir set.
+			Name:  "HOME",
+			Value: "/var/lib/rabbitmq",
+		},
+		{
+			// The various /usr/sbin/rabbitmq* scripts are really all the same
+			// wrapper shell-script that performs some "sanity checks" and then
+			// invokes the corresponding "real" program in
+			// /usr/lib/rabbitmq/bin.  The main "sanity check" is to ensure that
+			// the user running the command is either root or rabbitmq.  Inside
+			// of an openshift pod, however, the user is neither of these, so
+			// the wrapper script will always fail.
+
+			// By putting the real programs ahead of the wrapper in PATH we can
+			// avoid the unnecessary check and just run things directly as
+			// whatever user the pod has graciously generated for us.
+			Name:  "PATH",
+			Value: "/usr/lib/rabbitmq/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+		},
+	}
+
+	inetFamily := "inet"
+	inetProtocol := "tcp"
+	tlsArgs := ""
+	fipsArgs := ""
+	if IPv6Enabled {
+		inetFamily = "inet6"
+	}
+	erlangInetConfig := fmt.Sprintf("{%s,true}.\n", inetFamily)
+
+	if cluster.Spec.TLS.SecretName != "" {
+		inetProtocol = "tls"
+		tlsArgs = "-ssl_dist_optfile /etc/rabbitmq/inter-node-tls.config"
+		if fipsEnabled {
+			fipsArgs = "-crypto fips_mode true"
+		}
+	}
+
+	envVars = append(envVars, corev1.EnvVar{
+		Name: "RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS",
+		Value: fmt.Sprintf(
+			"-kernel inetrc '/etc/rabbitmq/erl_inetrc' -proto_dist %s_%s %s %s",
+			inetFamily,
+			inetProtocol,
+			tlsArgs,
+			fipsArgs,
+		),
+	}, corev1.EnvVar{
+		Name:  "RABBITMQ_CTL_ERL_ARGS",
+		Value: fmt.Sprintf("-proto_dist %s_%s %s", inetFamily, inetProtocol, tlsArgs),
+	})
+
+	defaultStatefulSet := rabbitmqv2.StatefulSet{
+		Spec: &rabbitmqv2.StatefulSetSpec{
+			Template: &rabbitmqv2.PodTemplateSpec{
+				EmbeddedObjectMeta: &rabbitmqv2.EmbeddedObjectMeta{},
+				Spec: &corev1.PodSpec{
+					SecurityContext: &corev1.PodSecurityContext{},
+					Containers: []corev1.Container{
+						{
+							// NOTE(gibi): if this is set according to the
+							// RabbitMQCluster name the the Pod will crash
+							Name:  "rabbitmq",
+							Image: cluster.Spec.Image,
+							Env:   envVars,
+							Args: []string{
+								// OSP17 runs kolla_start here, instead just run rabbitmq-server directly
+								"/usr/lib/rabbitmq/bin/rabbitmq-server",
+							},
+						},
+					},
+					InitContainers: []corev1.Container{
+						{Name: "setup-container", SecurityContext: &corev1.SecurityContext{}}},
+				},
+			},
+		},
+	}
+
+	if cluster.Spec.Override.StatefulSet == nil {
+		cluster.Spec.Override.StatefulSet = &defaultStatefulSet
+	}
+
+	if cluster.Spec.Override.StatefulSet.Spec.Template.Spec.NodeSelector == nil {
+		if nodeselector != nil {
+			cluster.Spec.Override.StatefulSet.Spec.Template.Spec.NodeSelector = *nodeselector
+		}
+	}
+
+	if topology != nil {
+		// Get the Topology .Spec
+		ts := topology.Spec
```

Comment:
should this be moved to `topology.ApplyTo(` pattern?

Reply:
cannot use it, different type https://github.com/openstack-k8s-operators/rabbitmq-cluster-operator/blob/v2.9.0_patches/api/v1beta1/rabbitmqcluster_types.go#L380